mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp

```diff
@@ -0,0 +1,136 @@
+{
+    g_pHudMenuStaticSystem->ReloadMenuList();
+}
+
+CON_COMMAND(mom_hud_menu_hide, "Hides the currently open hud menu if there is one.")
+{
+    g_pHudMenuStaticSystem->HideMenu();
+}
+
+static int HudMenuCompletion(const char *pPartial, char commands[COMMAND_COMPLETION_MAXITEMS][COMMAND_COMPLETION_ITEM_LENGTH]);
+
+CON_COMMAND_F_COMPLETION(mom_hud_menu_show, "Opens a hud menu. Accepts the hud menu itself by file name. See cfg/menus/", 0, HudMenuCompletion)
+{
+    if (args.ArgC() > 1)
+    {
+        g_pHudMenuStaticSystem->ShowMenu(args[1]);
+    }
+    else
+    {
+        Warning("This command expects a parameter that is the hud menu you want to open, by file name");
+    }
+}
+
+static int HudMenuCompletion(const char *pPartial, char commands[COMMAND_COMPLETION_MAXITEMS][COMMAND_COMPLETION_ITEM_LENGTH])
+{
+    const auto pCmdName = "mom_hud_menu_show";
+    char *pSubstring = nullptr;
+    if (Q_strstr(pPartial, pCmdName) && strlen(pPartial) > strlen(pCmdName) + 1)
+    {
+        pSubstring = (char *)pPartial + strlen(pCmdName) + 1;
+    }
+
+    const bool bSubstringNullOrEmpty = !pSubstring || !pSubstring[0];
+    int current = 0;
+
+    for (int i = 0; i < g_pHudMenuStaticSystem->GetNumMenus(); i++)
+    {
+        const char *menuName = g_pHudMenuStaticSystem->GetMenuName(i);
+
+        if (bSubstringNullOrEmpty || Q_stristr(menuName, pSubstring))
+        {
+            char command[COMMAND_COMPLETION_ITEM_LENGTH];
+            Q_snprintf(command, COMMAND_COMPLETION_ITEM_LENGTH, "%s %s", pCmdName, menuName);
+            Q_strncpy(commands[current], command, COMMAND_COMPLETION_ITEM_LENGTH);
+            current++;
+        }
+    }
+
+    return current;
+}
+
+CHudMenuStaticSystem::CHudMenuStaticSystem() : CAutoGameSystem("HudMenuStaticSystem") {}
+
+void CHudMenuStaticSystem::PostInit()
+{
+    ReloadMenuList();
+}
+
+void CHudMenuStaticSystem::ShowMenu(const char *pszFileName)
+{
+    CHudMenuStatic *pMenu = GET_HUDELEMENT(CHudMenuStatic);
+    if (!pMenu)
+        return;
+
+    if (!pMenu->IsMenuDisplayed())
+    {
+        pMenu->ShowMenu(pszFileName);
+        return;
+    }
+
+    pMenu->HideMenu();
+
+    if (!pMenu->IsMenuDisplayed(pszFileName))
+    {
+        pMenu->ShowMenu(pszFileName);
+    }
+}
+
+void CHudMenuStaticSystem::HideMenu() 
+{
+    CHudMenuStatic *pMenu = GET_HUDELEMENT(CHudMenuStatic);
+    if (!pMenu)
+        return;
+
+    if (pMenu->IsMenuDisplayed())
+    {
+        pMenu->HideMenu();
+    }
+}
+
+bool CHudMenuStaticSystem::IsMenuDisplayed(const char *pszMenuName)
+{
+    CHudMenuStatic *pMenu = GET_HUDELEMENT(CHudMenuStatic);
+    if (!pMenu)
+        return false;
+
+    return pMenu->IsMenuDisplayed(pszMenuName);
+}
+
+void CHudMenuStaticSystem::ReloadMenuList()
+{
+    m_vecMenuNames.RemoveAll();
+
+    FileFindHandle_t findHandle;
+    const char *pFilename = g_pFullFileSystem->FindFirstEx("cfg/menus/*.*", "MOD", &findHandle);
```

Comment:
They're all going to be res files so you'll want to instead do

```suggestion
    const char *pFilename = g_pFullFileSystem->FindFirstEx("cfg/menus/*.res", "MOD", &findHandle);
```

Reply:
I assume you mean vdf files. I changed it to explicitly reference them.